conjure-java-core/src/integrationInput/java/com/palantir/product/UnionTypeExample.java

```diff
@@ -124,6 +125,161 @@ public String toString() {
         T visitInterface(int value);
 
         T visitUnknown(String unknownType);
+
+        static <T> StringExampleStageVisitorBuilder<T> builder() {
+            return new VisitorBuilder<T>();
+        }
+    }
+
+    private static class VisitorBuilder<T>
+            implements StringExampleStageVisitorBuilder<T>,
+                    SetStageVisitorBuilder<T>,
+                    ThisFieldIsAnIntegerStageVisitorBuilder<T>,
+                    AlsoAnIntegerStageVisitorBuilder<T>,
+                    IfStageVisitorBuilder<T>,
+                    NewStageVisitorBuilder<T>,
+                    InterfaceStageVisitorBuilder<T>,
+                    UnknownStageVisitorBuilder<T>,
+                    CompletedStageVisitorBuilder<T> {
+        private Function<StringExample, T> stringExampleVisitor;
+
+        private Function<Set<String>, T> setVisitor;
+
+        private Function<Integer, T> thisFieldIsAnIntegerVisitor;
+
+        private Function<Integer, T> alsoAnIntegerVisitor;
+
+        private Function<Integer, T> ifVisitor;
+
+        private Function<Integer, T> newVisitor;
+
+        private Function<Integer, T> interfaceVisitor;
```

Comment:
Should we use IntFunction and DoubleFunction for int and double primitive types?

Reply:
Agree with carter here, we avoid boxing in other places and it would be nice to do so here as well